tests/flows/test_vesting.py

```diff
@@ -106,73 +103,535 @@ def test_main(
+            vesting_amount=initial_ocean_liq // 100 * 9,
+            vesting_blocks=100,
+            base_token_amount=initial_ocean_liq,
+            lp_swap_fee_amount=to_wei("0.003"),
+            publish_market_swap_fee_amount=to_wei("0.001"),
+            ss_contract=get_address_of_type(config, "Staking"),
+            base_token_address=ocean_contract.address,
+            base_token_sender=consumer_wallet.address,
+            publisher_address=consumer_wallet.address,
+            publish_market_swap_fee_collector=get_address_of_type(
+                config, "OPFCommunityFeeCollector"
+            ),
+            pool_template_address=get_address_of_type(config, "poolTemplate"),
+            from_wallet=consumer_wallet,
+        )
+        assert (
+            err.value.args[0]
+            == "execution reverted: VM Exception while processing transaction: revert ERC20Template: Vesting period too low. See FactoryRouter.minVestingPeriodInBlocks"
+        )
+
+
+@pytest.mark.unit
+def test_vesting_main_flow(web3, config, consumer_wallet, factory_router, side_staking):
+    """Tests consumer deploys pool, checks fees and checks if vesting is correct after 100 blocks."""
+
+    erc721_nft, erc20_token = deploy_erc721_erc20(
+        web3, config, consumer_wallet, consumer_wallet, cap=to_wei(1000)
+    )
+
+    # Consumer deploys pool & checks fees
+    mint_fake_OCEAN(config)
+    vesting_amount = to_wei(18)
+    initial_ocean_liq = to_wei(200)
+    pmt_collector = erc20_token.get_payment_collector()
+
+    block_before_deploying_pool = web3.eth.block_number
+    bpool = _deploy_ocean_pool(
+        erc20_token=erc20_token,
+        factory_router=factory_router,
+        config=config,
+        web3=web3,
+        from_wallet=consumer_wallet,
+    )
+    block_deployed_pool = block_before_deploying_pool + 2
+    assert bpool.is_finalized()
+    assert bpool.opc_fee() == to_wei("0.001")
+    assert bpool.get_swap_fee() == to_wei("0.003")
+    assert bpool.community_fee(get_address_of_type(config, "Ocean")) == 0
+    assert bpool.community_fee(erc20_token.address) == 0
+    assert bpool.publish_market_fee(get_address_of_type(config, "Ocean")) == 0
+    assert bpool.publish_market_fee(erc20_token.address) == 0
+    assert (
+        erc20_token.balanceOf(get_address_of_type(config, "Staking"))
+        == MAX_UINT256 - initial_ocean_liq
+    )
+
+    # check if the vesting amount is correct
+    assert side_staking.get_vesting_amount(erc20_token.address) == vesting_amount
+
+    # check if vesting is correct
+    dt_balance_before = erc20_token.balanceOf(erc20_token.get_payment_collector())
+    available_vesting_before = side_staking.get_available_vesting(erc20_token.address)
+
+    # advance 5 blocks to see if available vesting increased
+    for _ in range(4):
+        # send dummy transactions to increase block count (not part of the actual functionality)
+        send_ether(consumer_wallet, ZERO_ADDRESS, 1)
```

Comment:
You have 4 or 5 "advance N blocks" chunks of code, each doing the same thing except for N. DRY violation.

To fix: create a new method `_advanceBlocks(num_blocks:int)`, then make these chunks of code call that method instead.